db/db.go

```diff
@@ -312,14 +340,11 @@ func (b *boltDB) Commit(batch KVStoreBatch) (err error) {
 			break
```

Comment:
Should we clear the batch if the error is ErrAlreadyExist? I would say yes. So this could be a potential problem.

Reply:
i think it's ok, the purpose is that if any failure (regardless of AlreadyExist or other error) happens, the batch is not cleared for submitter to figure out what's wrong.

in our current code, if error does occur, we did not figure out or fix the batch, nor did we try re-submit. it only submit once, then the batch is never used any more.